src/datamodel/quests/quest_new.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::rc::Rc;
+
+#[derive(Debug, PartialEq)]
+pub enum Vote { Success, Failed, }
+
+pub struct QuestObjective {
+    pub num_players: u32,
+    pub required_successes: u32,
+}
+
+pub struct QuestNew {}
+
+impl QuestNew {
+    pub fn new(members: Vec<&'static str>, objective: Rc<QuestObjective>) -> OnGoingQuest {
+        let members: HashMap<&'static str, bool> = members.into_iter().map(|m| (m, false)).collect();
+        OnGoingQuest { 
+                objective,
+                members,
+                voting_counter: 0,
+                success_votes: 0,
+        }
+    }
+}
+
+pub struct OnGoingQuest {
+    members: HashMap<&'static str, bool>,
+    voting_counter: u32,
+    objective: Rc<QuestObjective>,
+    success_votes: u32,
+}
+
+impl OnGoingQuest {
+
+    pub fn add_vote(&mut self, player: &str, vote: Vote) {
+        if self.members[player] == true {
+            return;
+        }
+        self.voting_counter += 1;
+        if vote == Vote::Success {
+            self.success_votes += 1;
+        }
+        *self.members.get_mut(player).unwrap() = true;
+    }
+
+    pub fn finish_quest(self) -> Result<FinishedQuestNew, OnGoingQuest> {
+        let unvoted: Vec<bool> = self.members.iter().map(|m| *m.1).filter(|v| *v==false).collect();
+        if unvoted.len() > 0 {
+            Err(self)
+        } else {
+            if self.success_votes == self.objective.required_successes {
+                Ok( FinishedQuestNew { success: true, } )
+            }
+            else {
+                Ok( FinishedQuestNew { success: false, } )
+            }
+
+        }
+    }
+}
+
+pub struct FinishedQuestNew {
+    success: bool,
+}
+
+impl FinishedQuestNew {
+
+    pub fn result(&self) -> bool {
+        self.success
```

Comment:
If we do, we'll have to carry more information inside FinishedQuestNew's struct. In this particular case we'll have to add `objectives`, passing it previous state to this.
I thinks that `objectives` doesn't carry relevant information for the last state and to represent this state in the cleanest and more accurate way there must be only the result.
What do you think @julianmateu ?

Reply:
I think that keeping the objectives is useful for:
- querying the finished quest and see what happened
- associating the quest with the corresponding objective

I agree that this calculation should be done when creating the `FinishedQuestNew` object and stored in a field instead of calculating it every time the `result` method is called.

Reply:
`objectives` is a reference, so the upper scope will own it and it can keep track of it value and content. On the other hand, if we want to store Objectives and calculate just once `result` we'll need a constructor for `FinishedQuestNew` which add more code and turn more complex this implementation. I would rather keep it as simplest as possible knowing that further metadata can be added in the future to be return by result (i.e quest_number, quest_name, quest_win,...).

What do you think?

Reply:
I agree to keep it as it is for now